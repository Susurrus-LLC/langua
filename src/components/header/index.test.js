import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { PureHeader as Header } from '.'

describe('Header', () => {
  const siteData = {
    site: {
      siteMetadata: {
        title: 'Langua',
        subtitle: 'A suite of language tools'
      }
    }
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Header siteData={siteData} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately', () => {
    const tree = renderer.create(<Header siteData={siteData} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
