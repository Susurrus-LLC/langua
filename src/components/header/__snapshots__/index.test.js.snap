// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders accurately 1`] = `
<header
  className="header"
>
  <div
    className="titleArea"
  >
    <h1
      className="siteTitle"
    >
      <a
        className="titleLink"
        href="/"
      />
    </h1>
    <p
      className="subtitle"
    />
  </div>
  <nav
    className="nav"
  >
    <ul
      className="menu"
    >
      <li
        className="navLink"
      >
        <a
          href="/gen"
        >
          Gen
        </a>
      </li>
      <li
        className="navLink disabled"
      >
        <a
          href="/morph"
        >
          Morph
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          href="/deriv"
        >
          Deriv
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          href="/frequen"
        >
          Frequen
        </a>
      </li>
      <li
        className="navLink disabled"
      >
        <a
          href="/tree"
        >
          Tree
        </a>
      </li>
      <li
        className="navLink disabled"
      >
        <a
          href="/lex"
        >
          Lex
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
